src/main/java/me/botsko/prism/actions/ItemStackAction.java

```diff
@@ -42,633 +36,576 @@
+            final BookMeta bookMeta = (BookMeta) meta;
+            bookMeta.setAuthor(actionData.by);
+            bookMeta.setTitle(actionData.title);
+            bookMeta.setPages(actionData.content);
+            item.setItemMeta(bookMeta);
+        }
+        if (meta instanceof FireworkEffectMeta && actionData.effectColors != null
+                && actionData.effectColors.length > 0) {
+            final FireworkEffectMeta fireworkMeta = (FireworkEffectMeta) meta;
+            final Builder effect = FireworkEffect.builder();
+
+            for (int i = 0; i < actionData.effectColors.length; i++) {
+                effect.withColor(Color.fromRGB(actionData.effectColors[i]));
+            }
+            fireworkMeta.setEffect(effect.build());
+
+            if (actionData.fadeColors != null) {
+                for (int i = 0; i < actionData.fadeColors.length; i++) {
+                    effect.withFade(Color.fromRGB(actionData.fadeColors[i]));
+                }
+                fireworkMeta.setEffect(effect.build());
+            }
+            if (actionData.hasFlicker) {
+                effect.flicker(true);
+            }
+            if (actionData.hasTrail) {
+                effect.trail(true);
+            }
+            fireworkMeta.setEffect(effect.build());
+            item.setItemMeta(fireworkMeta);
+        }
+
+        if (actionData.name != null) {
+            if (meta == null) {
+                meta = item.getItemMeta();
+            }
+
+            if (meta != null) {
+                meta.setDisplayName(actionData.name);
+            }
+        }
+
+        if (actionData.lore != null) {
+            if (meta == null) {
+                meta = item.getItemMeta();
+            }
+
+            if (meta != null) {
+                meta.setLore(Arrays.asList(actionData.lore));
+            }
+        }
+
+        if (meta != null) {
+            item.setItemMeta(meta);
+        }
+    }
+
+    public ItemStackActionData getActionData() {
+        return this.actionData;
+    }
+
+    /**
+     * ItemStack.
+     * @return ItemStack
+     */
+    public ItemStack getItem() {
+        return item;
+    }
+
+    /**
+     * Nice name.
+     * @return String
+     */
+    @Override
+    public String getNiceName() {
+        String name = "";
+        if (item != null) {
+            final String fullItemName = ItemUtils.getItemFullNiceName(item);
+            name = actionData.amt + " " + fullItemName;
+        }
+        return name;
+    }
+
+    @Override
+    public ChangeResult applyRollback(Player player, QueryParameters parameters, boolean isPreview) {
+        return placeItems(player, parameters, isPreview);
+    }
+
+    @Override
+    public ChangeResult applyRestore(Player player, QueryParameters parameters, boolean isPreview) {
+        return placeItems(player, parameters, isPreview);
+    }
+
+    protected ChangeResult placeItems(Player player, QueryParameters parameters, boolean isPreview) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'placeItems(Player, QueryParameters, boolean)' has an NPath complexity of 20409384](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)